src/btree/row_modify.c

```diff
@@ -29,7 +29,7 @@ __wt_page_modify_alloc(WT_SESSION_IMPL *session, WT_PAGE *page)
      * another thread did the work.
      */
     if (__wt_atomic_cas_ptr(&page->modify, NULL, modify))
-        __wt_cache_page_inmem_incr(session, page, sizeof(*modify));
```

Comment:
Given we do an `atomic_cas`, we should not double count, we are already safe on that side. Should we set `new_update` to true since we are allocating memory for the page's modification structure?

Reply:
No because it's a shared structure.
It's easier to think in terms "what's going to be released once the transaction is committed or rolled back while all others stay?"
The page's modification only gets released on page_out.